src/transactions/OfferExchange.cpp

```diff
@@ -1259,14 +1272,26 @@ exchangeWithPool(int64_t reservesToPool, int64_t maxSendToPool, int64_t& toPool,
         bool res = hugeDivide(fromPool, maxBps - feeBps,
                               bigMultiply(reservesFromPool, toPool),
                               denominator, ROUND_DOWN);
-        if (res)
-        {
-            fromPool = std::min(fromPool, maxReceiveFromPool);
-        }
-        return res;
+
+        // Fail if the division overflows, or if we would be receiving too much
+        // from the pool.
+        return res && fromPool <= maxReceiveFromPool;
```

Comment:
Is this just a sanity check? Since you made [this comment](https://github.com/stellar/go/pull/3888#discussion_r703797219) which proves this is guaranteed not to happen?

Reply:
Yeah I'm **pretty** sure you're right. We should throw if `fromPool > maxReceiveFromPool`.